### PR TITLE
Cwltool parsing and API

### LINKED ISSUE #70: Display EDAM Ontology information
Most commonly found in File format fields

### LINKED ISSUE #94: Support inline tools in `run` parameters
eg https://github.com/common-workflow-language/common-workflow-language/blob/master/v1.0/examples/nestedworkflows.cwl#L37

### LINKED ISSUE #115: Better details about parsing errors
My CWL workflow (a pretty simple one, I think: https://github.com/ICGC-TCGA-PanCancer/OxoG-Dockstore-Tools/blob/develop/zip_and_index_vcf.cwl) failed with parsing error:

> Error: The workflow could not be parsed from the given URL

This doesn't give much explanation as to what the parsing error was or how to fix it. A little more detail would be nice.

Also, if I give it the URL to a CommandLineTool, it fails with a parsing error. I didn't really expect a diagram from a Tool, but I think a message saying "This viewer only accepts URLs for Workflows" would have made this type of error more clear.

### LINKED ISSUE #116: Slashes in a branch name cause problems with URL extraction
Currently the URL is split by slashes which causes issues in workflows such as:

```
http://view.commonwl.org/workflows/github.com/ICGC-TCGA-PanCancer/OxoG-Dockstore-Tools/blob/feature/fix_annnotator_steps/zip_and_index_vcf.cwl
```

Where the branch is determined to be `feature` and not `feature/fix_annotator_steps`

### LINKED ISSUE #126: Add documentation for best practices
* Information on what fields are preferable to include to make workflows look good in the visualisation and tables
* Best practices referencing http://www.commonwl.org/user_guide/rec-practices/
* Explanation of Research Objects and how the bundles are constructed/which metadata is added
* Any current quirks/limitations of CWLViewer and how to mitigate them, along with Github issues referencing improvements for these

### LINKED ISSUE #132: Provide preferred citation for the CWL viewer
Embed in the generated RO as well?

I've found the following DOIs -- which is preferred?

https://doi.org/10.5281/zenodo.823535

https://doi.org/10.7490/f1000research.1114375.1

### PR DESCRIPTION
**Pull request by MarkRobbo**
This PR is to track progress on current development and issues closed in the dev environment

* Adds cwltool use for parsing - uses RDF representation
* Triple store to collect workflow details
* Intermediate loading screen
* Research Object enhancements
    - Add visualisation images as bundled aggregates
    - RDF representation as annotation
    - Packed version of workflow files as annotation
    - Git2prov link for the repository as history
* API and documentation backing current features
* Visualisation improvements
    - Stacking of default values
* Ontology information linked and name given for `format` fields on inputs and outputs
* "About" page with best practices for writing CWL for parsing by the viewer
* Various bug fixes

Still needs before merge
- [x] Revise caching strategy and code
- [x] Apache Fuseki requirement in Dockerfile
- [x] Test coverage of queued workflows
- [x] Test coverage of API

Breaks private repository access through Github

Closes #116 
Closes #115 
Closes #94 
Closes #70 
Closes #126 
Closes #132

### CODE REVIEW

File: src/main/resources/templates/about.html [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+                over the top level tool doc. You can use this to provide a more descriptive label of the tool's
+                application in the particular step if preferred</p>
+
+            <h2>Conceptual Identifiers</h2>
+            <p class="recommendation">All <code>input</code> and <code>output</code> identifiers should reflect their conceptual identity.
+               Generic and uninformative names such as <code>result</code> or <code>input</code>/<code>output</code>
+               should be avoided</p>
+            <p class="why">Helpful identifiers allow for the links between steps in the CWL file to be easily distinguished</p>
+            <p class="use">Identifiers are displayed in the tables and are unique to the step. If <code>label</code> or
+               <code>doc</code> strings are for some reason not provided, the IDs will be used in the visualisation</p>
```
Comment (mr-c) [inline src/main/resources/templates/about.html]: This implies that the `label` or `doc` shows up in the graph -- I know you mean in the listing below, but that could be clarified.

Comment (MarkRobbo) [inline src/main/resources/templates/about.html]: The `label` is used in the graph as a change to the current production version. The `doc` is not, I will certainly reword this so it is more clear.

Thank you for the feedback. Agree on the maintenance burden but it also acts somewhat as documentation for the features and what people can use within CWLViewer, so I think it is acceptable. I also plan to reference the official best practice document where relevant, once it is live.

File: src/main/resources/templates/about.html [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+            <div style="position:relative;float:right;height:0;padding-bottom:30%;width:50%;margin-left:20px;">
+                <iframe src="https://www.youtube-nocookie.com/embed/_yjhVTmvxLU?rel=0?ecver=2" style="position:absolute;width:100%;height:100%;left:0" allowfullscreen="true"></iframe>
+            </div>
+
+            <p>CWL Viewer is a richly featured web visualisation suite for workflows written in the
+                <a href="http://www.commonwl.org/">Common Workflow Language</a> with an aim of facilitating sharing,
+                understanding and discovery as well as encouraging best practices when writing workflows and their
+                tooling.</p>
+
+            <p>Cite as: <code><a href="https://doi.org/10.7490/f1000research.1114375.1">10.7490/f1000research.1114375.1</a></code></p>
```
Comment (mr-c) [inline src/main/resources/templates/about.html]: "Please". Consider following the Crossref DOI display guidelines: https://www.crossref.org/display-guidelines/

File: src/main/resources/templates/about.html (hunk 1 above)
Comment (mr-c) [inline src/main/resources/templates/about.html]: Cool! You are welcome.

**Pull request merged**